development/tools/cmd/pjtester/README.md

```diff
@@ -0,0 +1,93 @@
+# ProwJob tester
+
+## Overview
+
+ProwJob tester is a tool for testing changes to ProwJobs and scripts in the `test-infra` repository which are under development. It uses the production Prow instance to run chosen ProwJobs with code from pull requests (PRs) without going through multiple cycles of new PRs, reviews, and merges. The whole development is done within one PR.
+
+### How it works
+
+The workhorse for testing ProwJobs is a tool written in Go called `pjtester`. It's available in the `prow-tools` Docker image.
+
+`pjtester` is executed by the presubmit job `pre-master-test-infra-pjtester`. This presubmit job is triggered when something changes under the `test-infra` repository virtual path `vpath/pjtester.yaml`. 
+It is configured by the **run-if-changed** option:
+```bash
+run_if_changed: "^(vpath/pjtester.yaml)"
+```
+
+`pjtester` expects to find the the configuration of the ProwJobs to tests under `vpath/pjtester.yaml`.
+
+By default, `pjtester` disables all reporting for the ProwJob. That means no Slack messages and no status report on Github. To check the test results, consult the [ProwStatus](https://status.build.kyma-project.io/) dashboard.
+
+Details from `pjtester.yaml` and from the ProwJob environment variables are used to construct the specification of the ProwJob to test. `pjtester` uses the environment variables created by Prow for the presubmit job which identify the pull request and its commit hash in the `test-infra` repository. The generated ProwJob to test then uses the `test-infra` code from the pull request's head, ensuring that the latest code is under test.
```

Comment:
`which identify` - does this refer to the variables or the presubmit job?

Reply:
Environment variables existing in presubmit holds data which identify pull request for which this presubmit is running.

Reply:
Great, thanks for the clarification! All good then!